webclient/app/src/app/commons/templateDialog/TemplateDialog.jsx

```diff
@@ -0,0 +1,65 @@
+import React, {useEffect} from "react";
+import {
+    Button,
+    Box,
+    Dialog,
+    DialogActions,
+    DialogTitle,
+    IconButton,
+    TextField,
+    Typography
+} from "@mui/material";
+import LoadingSpinner from "../loadingSpinner/LoadingSpinner";
+import TemplateDialogStyles from "./TemplateDialogStyles";
+import {useTranslation} from "react-i18next";
+import {Close} from "@mui/icons-material";
+
+function TemplateDialog(props) {
+
+    let [appTemplate, setAppTemplate] = React.useState(null);
+    const templateDialogStyles = TemplateDialogStyles();
+    const {t} = useTranslation();
+
+    const {appTemplateId, onClose, handleSave, appTemplates} = props;
+
+    useEffect(() => {
+        setAppTemplate({...appTemplates.find(appTemplate => appTemplate.id === appTemplateId)})
+    }, [appTemplateId, appTemplates])
+    
+    if (!appTemplate) {
+        return <LoadingSpinner message={t("appTemplateDialog.isLoading")}/>
+    }
+
+    return (
+        <Dialog open={appTemplateId} maxWidth={"xm"} spacing={2}>
+            <DialogTitle className={templateDialogStyles.dialogHeaderBar}>
+                <Typography noWrap variant={"h6"} component={"p"}>{appTemplate.name + " " + t("appTemplateDialog.edit")}</Typography>
```

Comment:
This translation could be dangerous because some languages have different grammar than german/english. See Interpolation on i18next (https://www.i18next.com/translation-function/interpolation) to learn more